scout/adapter/mongo/variant.py

```diff
@@ -469,69 +471,94 @@ def _find_affected(self, case_obj):
 
         return affected_ids
 
-    def match_affected_gt(self, case_obj, institute_obj, positional_variant_ids, limit_genes):
+    def match_affected_gt(self, case_obj, positional_variant_ids, limit_genes):
         """Match positional_variant_ids against variants from affected individuals
         in a case, ensuring that they at least are carriers.
 
         Args:
             case_obj (dict): A Case object.
-            institute_obj (dict): An Institute object.
             positional_variant_ids (iterable): A set of possible variant_ids (md5 key based upon a list like this: [ "17", "7577559", "G" "A", "research"]
             limit_genes (list): list of gene hgnc_ids to limit the search to
 
         Returns:
-            causatives(iterable(Variant))
+            (iterable(Variant))
         """
 
-        if len(positional_variant_ids) == 0:
-            return []
-
         filters = {"variant_id": {"$in": list(positional_variant_ids)}}
-        if case_obj:
-            affected_ids = self._find_affected(case_obj)
-            if len(affected_ids) == 0:
-                return []
-
-            filters["case_id"] = case_obj["_id"]
-            filters["samples"] = {
-                "$elemMatch": {
-                    "sample_id": {"$in": affected_ids},
-                    "genotype_call": {"$regex": "1"},
-                }
+
+        affected_ids = self._find_affected(case_obj)
+        if len(affected_ids) == 0:
+            return []
+        filters["case_id"] = case_obj["_id"]
+        filters["samples"] = {
+            "$elemMatch": {
+                "sample_id": {"$in": affected_ids},
+                "genotype_call": {"$regex": CARRIER},
             }
-        else:
-            filters["institute"] = institute_obj["_id"]
+        }
+
         if limit_genes:
             filters["genes.hgnc_id"] = {"$in": limit_genes}
 
         return self.variant_collection.find(filters)
 
-    def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
-        """Check if there are any variants that are previously marked causative
+    def institute_causatives(self, institute_obj, limit_genes=None):
```

Comment:
This is a completely new function called from institute controllers to get variants for the Cauaatives page